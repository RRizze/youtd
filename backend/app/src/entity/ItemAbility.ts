import { Entity, PrimaryColumn } from "typeorm";

@Entity()
export class ItemAbility {
  @PrimaryColumn()
  abilityId: number;

  @PrimaryColumn()
  itemId: number;
}