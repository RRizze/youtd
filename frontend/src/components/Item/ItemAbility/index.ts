export { default } from "./item-ability";
