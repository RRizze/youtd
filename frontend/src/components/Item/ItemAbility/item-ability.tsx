import React from "react";
import "./style.scss";
import { IAbility } from "../../../models/item";

const ItemAbility = ({
  id,
  ability_name,
  ability_description,
  lvl_bonus,
  ability_active,
}: IAbility) => {
  let bonus;
  let active;

  if (typeof lvl_bonus === "object") {
    bonus = <ul>
      {lvl_bonus.map((b, idx) => (
        <li key={b + idx}>b</li>
      ))}
    </ul>;
  } else {
    bonus = <p>{lvl_bonus}</p>;
  }

  if (typeof ability_active === "object") {
    active = <ul>
      {ability_active.map((a, idx) => (
        <li key={a + idx}>{a}</li>
      ))}
    </ul>;
  } else {
    active = <p>{ability_active}</p>;
  }

  return (
    <React.Fragment>
      <h3 className="ability__title">{ability_name}</h3>
      <p className="ability__description">{ability_description}</p>
      {bonus}
      {active}
    </React.Fragment>
  );
};

export default ItemAbility;
