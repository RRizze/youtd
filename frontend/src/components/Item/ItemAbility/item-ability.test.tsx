import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import ItemAbility from "./item-ability";

describe("ItemAbility tests", () => {
  it("render correctly", () => {
    const props = {
      id: 1,
      ability_name: "dagger",
      ability_description: "desc",
      lvl_bonus: "lvl bonus",
      ability_active: "ability active",
    };
    const tree = renderer
      .create(<ItemAbility {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render with props", () => {
    const props = {
      id: 1,
      ability_name: "dagger",
      ability_description: "desc",
      lvl_bonus: "lvl bonus",
      ability_active: "ability active",
    },
      wrapper = mount(<ItemAbility {...props} />);

    expect(wrapper.props().ability_name).toEqual("dagger");
  });
});
