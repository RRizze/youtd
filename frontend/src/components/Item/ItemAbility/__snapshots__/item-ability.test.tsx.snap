// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemAbility tests render correctly 1`] = `
Array [
  <h3
    className="ability__title"
  >
    dagger
  </h3>,
  <p
    className="ability__description"
  >
    desc
  </p>,
  <p>
    lvl bonus
  </p>,
  <p>
    ability active
  </p>,
]
`;
